src/fs/cluster_info.hh

```diff
@@ -0,0 +1,76 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright (C) 2020 ScyllaDB
+ */
+
+#pragma once
+
+#include "fs/cluster.hh"
+#include "fs/inode.hh"
+#include "fs/units.hh"
+#include <cassert>
+#include <map>
+
+namespace seastar::fs {
+
+class cluster_info {
+
+    // Each cluster_data_vec corresponds to an inode_data_vec
+    struct cluster_data_vec {
+        inode_t data_owner;
+        file_range data_range;
+    };
+    size_t valid_data_size = 0;
+    bool read_only = false;
+    std::map<disk_offset_t, cluster_data_vec> data;
```

Comment:
In compaction we will need to read `data` so we will need getter for that. Also `cluster_data_vec` should be public then.